### PR TITLE
Update phone number modal

### PR DESCRIPTION
**Pull request by cisel**
# Description

# Checklist

Pull request contains:

- [ ] A new component
- [ ] Component maintenance: improvement / bug fix / etc
- [ ] Component library change: storybook / webpack / etc

Definition of done:

- [ ] Includes theme changes for both Uswitch and money
- [ ] Work has been tested in multiple browsers
- [ ] PR description includes description of change
- [ ] PR description includes screenshot of change
- [ ] If new component, designer has approved screenshot
- [ ] If the change will affect other teams, that team knows about this change
- [ ] If introducing a new component behaviour, added a story to cover that case.

### CODE REVIEW

File: src/compounds/phone-number-modal/src/index.tsx [hunk 1]
```diff
@@ -1,154 +1,177 @@
...
-  complianceText: any
+  complianceText: string[]
+}
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+  phoneNumberModalInfo: PhoneNumberModalInfo
+  logoImageCritical?: boolean
+  isOpen: boolean
+  modalAriaLabel: string
+  setStateClosed: setStateClosed
```
Comment (samhstn) [inline src/compounds/phone-number-modal/src/index.tsx:24]:
```javascript
setStateClosed: () => void
```

or update the type to be capitalized (`SetStateClosed`)

File: src/compounds/phone-number-modal/src/index.tsx [hunk 2]
```diff
@@ -1,154 +1,177 @@
...
+  const complianceTextElements = phoneNumberModalInfo.complianceText.map(
+    (text, i) => {
+      return (
+        <p
+          key={`compliance_text_${i}`}
+          sx={{
+            fontWeight:
+              i === phoneNumberModalInfo.complianceText.length - 1
+                ? 'bolder'
+                : 'base'
```
Comment (samhstn) [inline src/compounds/phone-number-modal/src/index.tsx:43]: Why is the last `p` tag bold? Maybe it will make sense from screenshots

**Review by samhstn — COMMENTED** [review]
Code looks good @cisel 👍 

Could you add screenshots with money theme?
(would help to understand why compliance text is bold)

File: src/compounds/phone-number-modal/src/index.tsx (hunk 1 above)
Comment (cisel) [reply src/compounds/phone-number-modal/src/index.tsx]: Changed this, thanks.

**Review by samhstn — APPROVED** [review]

**Review by samhstn — APPROVED** [review]

**Pull request merged**